src/ucp/wireup/wireup_cm.c

```diff
@@ -243,10 +244,186 @@ ucs_status_t ucp_ep_client_cm_connect_start(ucp_ep_h ucp_ep,
+    ucp_conn_request_h ucp_conn_request;
+    ucs_status_t status;
+
+    ucp_conn_request = ucs_malloc(ucs_offsetof(ucp_conn_request_t, sa_data) +
+                                  remote_data->conn_priv_data_length,
+                                  "ucp_conn_request_h");
+    if (ucp_conn_request == NULL) {
+        ucs_error("failed to allocate connect request, rejecting connection request %p on TL listener %p",
+                  conn_request, listener);
+        goto err_reject;
+    }
+
+    ucp_conn_request->remote_dev_addr = ucs_malloc(remote_data->dev_addr_length,
+                                                   "remote device address");
+    if (ucp_conn_request->remote_dev_addr == NULL) {
+        ucs_error("failed to allocate device address, rejecting connection request %p on TL listener %p",
+                  conn_request, listener);
+        goto err_free_ucp_conn_request;
+    }
+
+    ucp_conn_request->listener     = ucp_listener;
+    ucp_conn_request->uct.listener = listener;
+    ucp_conn_request->uct_req      = conn_request;
+    strncpy(ucp_conn_request->dev_name, local_dev_name, UCT_DEVICE_NAME_MAX);
+    ucp_conn_request->dev_name[UCT_DEVICE_NAME_MAX - 1] = '\0';
+    memcpy(ucp_conn_request->remote_dev_addr, remote_data->dev_addr,
+           remote_data->dev_addr_length);
+    memcpy(&ucp_conn_request->sa_data, remote_data->conn_priv_data,
+           remote_data->conn_priv_data_length);
+
+    uct_worker_progress_register_safe(ucp_listener->worker->uct,
+                                      ucp_cm_server_conn_request_progress,
+                                      ucp_conn_request,
+                                      UCS_CALLBACKQ_FLAG_ONESHOT, &prog_id);
+
+    /* If the worker supports the UCP_FEATURE_WAKEUP feature, signal the user so
+     * that he can wake-up on this event */
+    ucp_worker_signal_internal(ucp_listener->worker);
+    return;
+
+err_free_ucp_conn_request:
+    ucs_free(ucp_conn_request);
+err_reject:
+    status = uct_listener_reject(listener, conn_request);
+    if (status != UCS_OK) {
+        ucs_warn("failed to reject connect request %p on listener %p",
+                 conn_request, listener);
+    }
+}
+
+ucs_status_t
+ucp_ep_cm_server_create_connected(ucp_worker_h worker,
+                                  const ucp_ep_params_t *params,
+                                  const ucp_unpacked_address_t *remote_addr,
+                                  ucp_ep_h *ep_p)
+{
+    unsigned ep_init_flags = ucp_cm_ep_init_flags(worker, params);
+    char peer_name[UCS_SOCKADDR_STRING_LEN];
+    ucp_ep_h ep;
+    ucp_lane_index_t lane_idx;
+    ucs_status_t status;
+
+    UCS_STATIC_ASSERT(UCS_SOCKADDR_STRING_LEN <= UCP_WORKER_NAME_MAX);
+
+    ucs_assert(params->field_mask & (UCP_EP_PARAM_FIELD_SOCK_ADDR |
+                                     UCP_EP_PARAM_FIELD_CONN_REQUEST));
+    if (params->field_mask & UCP_EP_PARAM_FIELD_SOCK_ADDR) {
```

Comment:
why would server side expect UCP_EP_PARAM_FIELD_SOCK_ADDR?
shouldn't it be always UCP_EP_PARAM_FIELD_CONN_REQUEST and that's it?